test/test_tensorflow.py

```diff
@@ -57,29 +64,39 @@ def test_horovod_allreduce_cpu(self):
         """Test on CPU that the allreduce correctly sums 1D, 2D, 3D tensors."""
         hvd.init()
         size = hvd.size()
-        with self.test_session(config=self.config) as session:
-            dtypes = [tf.int32, tf.int64, tf.float16, tf.float32, tf.float64]
-            dims = [1, 2, 3]
-            for dtype, dim in itertools.product(dtypes, dims):
-                with tf.device("/cpu:0"):
-                    tf.set_random_seed(1234)
-                    tensor = tf.random_uniform(
-                        [17] * dim, -100, 100, dtype=dtype)
-                    summed = hvd.allreduce(tensor, average=False)
-                multiplied = tensor * size
-                max_difference = tf.reduce_max(tf.abs(summed - multiplied))
+        dtypes = [tf.int32, tf.int64, tf.float16, tf.float32, tf.float64]
+        dims = [1, 2, 3]
+
+        def graph_and_eager(dtype, dim):
```

Comment:
There's an annotation you can use to accomplish much the same thing:

```
from tensorflow.python.framework.test_util import run_in_graph_and_eager_modes
```

Reply:
thanks. i changed to use run_in_graph_and_eager_modes.